src/main/java/com/google/firebase/auth/FirebaseAuth.java

```diff
@@ -53,7 +57,12 @@ public TenantManager get() {
   }
 
   public TenantManager getTenantManager() {
-    return tenantManager.get();
+    TenantManager suppliedTenantManager;
+    synchronized (lock) {
+      suppliedTenantManager = tenantManager.get();
+      tenantManagerCreated.set(true);
```

Comment:
Set this to true in the `Supplier.get()` method (in the FirebaseAuth constructor). That method gets further wrapped in a `threadSafeMemoize` which provides the necessary locking around it.

Reply:
Oh great idea! This looks much cleaner.